src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1141,6 +1141,9 @@ public static IntPtr GetFunctionPointerForDelegate(Delegate d)
         {
             ArgumentNullException.ThrowIfNull(d);
 
+            if (OperatingSystem.IsWasi() || OperatingSystem.IsBrowser())
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_DynamicEntrypoint);
```

Comment:
we should update src/libraries/System.Runtime/ref/System.Runtime.cs to add `UnsupportedOSPlatform` for browser/wasi to GetFunctionPointerForDelegate